# Momentum 26-Oct-2015

In a previous workbook (http://pdf.markcarter.me.uk/momo2014-11-26a.html), we asked: **what is the expected of investing in each decile of the relative strength of shares over 6 months over the subsequent 6 months?**

We saw that it paid to invest in the best-performing decile, and avoid the worst-performing one. We repeat the analysis for data collect near 26-Oct-2015, **and we come to the same conclusion**.

In [1]:
%matplotlib inline
import matplotlib.mlab as mlab
import matplotlib.pyplot as plt
import pylab
import numpy as np
import pandas as pd

In [2]:
import mython
import mython.sharelock
import mython.quant as quant

## Prep the data

You won't have to do this, normally, it is just a data preparation stage. There are two ways of doing it:

*Method 1*

In [3]:
%%bash
mython sharelock momo

Namespace(momo='momo')
Finished


*Method 2*

In [4]:
mython.sharelock.momo()

The data would then have to be uploaded to pastebin: 

http://pastebin.com/raw.php?i=xp6ZfCkv

## Thresholds for deciles

Load the full dataset, and build a dataframe of just the data we need:

In [5]:
inp = pd.io.parsers.read_csv("http://pastebin.com/raw.php?i=xp6ZfCkv")
df = inp[['epic', 'rs6mb', 'rs1y']].copy()
df['rs6ma'] = (df.rs1y/100.0 + 1.0)/(df.rs6mb/100.0 + 1.0)*100.0 - 100.0
df.head(5)

epic  rs6mb   rs1y      rs6ma
0   3IN  17.72  18.50   0.662589
1   888  20.21  24.03   3.177772
2   AA. -28.56 -15.07  18.882979
3   AAL -35.11 -55.84 -31.946371
4  ABBY  36.05  26.04  -7.357589

Here are the decile ranges for the first 6 months of the loaded data:

In [6]:
quant.quantiles(df.rs6ma, 10)

0.0    -57.507175
0.1    -22.749957
0.2    -12.785435
0.3     -7.433671
0.4     -2.383119
0.5      1.964370
0.6      6.739675
0.7     12.038106
0.8     18.667706
0.9     28.594087
1.0    173.164025
Name: rs6ma, dtype: float64

As you can see, the worst-performing share had an RS (relative strength) of -57.5%, whilst the best was 173.2%. 10% of the shares had a RS of at least 28.6%

## Subsequent performance

Let's see how the companies for the various buckets in the first 6 months performed over the subsequent 6 months:

In [7]:
def agg(df, n):
    df['tile'] = pd.qcut(df['rs6ma'], n, labels = range(n))
    res = df.groupby('tile')['rs6mb'].mean()
    return res

agg(df, 10)

tile
0    -1.691194
1     6.492576
2    12.801970
3    14.193636
4    13.264545
5    13.965000
6     9.406818
7    14.974091
8    10.214091
9    17.115970
Name: rs6mb, dtype: float64

The table above gives the *expected* subsequent 6 month return.

You may be asking yourself how it is possible for most of the deciles are positive. Surely there should be some cancelling out? The answer is that the relative strengths are relative to the Footsie, so if the larger universe consistently outperforms the Footsie, then you would expect the deciles to do so, too.

## Conclusion

The conclusion obtained for the momentum experiment of 26-Nov-2014 still hold true: the best-performing decile for the first 6 months went on to be the best performing decile for the next 6 months. The worst-performing decile went on to perform the worst.

**Momentum effects are strong**.

## About this document

    Author:  Mark Carter
    Created: 26-Oct-2015

The dataset was obtained at about this document creation date.